### DISCUSSION TITLE
Test Cases Basic and Enhanced

### DISCUSSION THREAD

**Discussion by FabrizioMoggio** [camaraproject/CallForwardingSignal#125] [General]
Considering: https://github.com/camaraproject/CallForwardingSignal/issues/124

Considering the structure of the readiness CL table: https://lf-camaraproject.atlassian.net/wiki/spaces/CAM/pages/14559630/API+Release+Process#Checklist-explanation

I'm wondering if should split the Gherkin files in "basic" (sunny days) for row 7 and "Enhanced" (rainy days) for row 8.
I could split, for example, the current "call-forwarding-signal-every-forwarding.feature" into "basic-call-forwarding-signal-every-forwarding.feature" and "enhanced-call-forwarding-signal-every-forwarding.feature".

What do you think about?

@bigludo7 maybe some new test case can be added in the "Enhanced". You had some idea for those test cases if I'm not wrong. 
@StefanoFalsetto-CKHIOD any "Enhanced" or "Basic" test case from your experience in the development?

**Comment by bigludo7** [comment]
Hello @FabrizioMoggio 
To be sure I got it right:  we need the Enhanced TC if we target a stable version and as of now we're not in position to target stable as we have no implementation listed here https://open-gateway.gsma.com/ - correct?

For your proposal on having 2 files I've no preference but as Toyeeb Rehman & Ola Ajibola as GSMA certification owners are 'customer' of these files we can ask them feedback.

**Reply by FabrizioMoggio** [reply]
I was actually thinking to target a Stable version (v1.0.0) as outcome of a development phase (from CK Hutchison). My understanding was that a Stable version goes to GSMA for certification and, when commercially exposed, goes to that portal.

If I got you right, you are saying that an API becomes Stable only AFTER the certification from GSMA?

**Comment by StefanoFalsetto-CKHIOD** [comment]
Hi @bigludo7 CKH finished the implementation and is waiting for GSMA to update the open gateway website. We will likely see this update by the end of this week.

**Reply by FabrizioMoggio** [reply]
About the two files, my proposal is only to be able to fill both row 7 and 8 of the RCL. Otherwise how can we do it?

**Reply by bigludo7** [reply]
@StefanoFalsetto-CKHIOD Great ! So I guess we can target Stable.

For the 2 rows @FabrizioMoggio we can direct to the same file. This is what we did for the V1 version of the first meta release.

About the test definition themselves we have to provide more depth in the test and for example cover error scenario and corner UC (if we have). We can take a look on Test scope for other v1.0 API and be sure we have same coverage.

**Comment by FabrizioMoggio** [comment]
Let's continue any further discussion on "Stable" here: https://github.com/camaraproject/CallForwardingSignal/discussions/115

**Comment by FabrizioMoggio** [comment]
About the split of the testing files in Basic and Enhanced, I suggest to to see the outcome of https://github.com/camaraproject/CallForwardingSignal/issues/124

**Comment by FabrizioMoggio** [comment]
Dear @bigludo7 considering your suggestion on asking about the split of the Gherkin files to GSMA. I'm wondering if this is the right moment considering they are currently certifying v0.2.0. I don't want to slow dawn the process creating some doubt. @StefanoFalsetto-CKHIOD what do you think about?

 So I suggest to start asking to Tanja.

**Reply by bigludo7** [reply]
Hello @FabrizioMoggio On this specific topic for one or two file I've no preference. I fully support both option.

**Comment by FabrizioMoggio** [comment]
According to the new note in column 8: https://lf-camaraproject.atlassian.net/wiki/spaces/CAM/pages/14559630/API+Release+Process#Checklist-explanation

We can refer to the same Gherkin files in both 7 and 8.

**Comment by FabrizioMoggio** [comment]
Implemented
